usr/sbin/regionsrv-enabler-azure

```diff
@@ -12,6 +12,34 @@
 # Lesser General Public License for more details.
 # You should have received a copy of the GNU Lesser General Public
 # License along with this library.
+
+"""
+Check the different states of the license type:
+    START   LICENSE CHANGED TO     CACHE   ACTION
+    BYOS-1  SLES                   N/A     Register
+    BYOS-1  SLES                   BYOS    Register
+    BYOS-1  BYOS-1                 N/A     Do nothing
+    BYOS-1  BYOS-1                 BYOS    Do nothing
+    BYOS-1  BYOS-2                 N/A     Do nothing
+    BYOS-1  BYOS-2                 BYOS    Do nothing
+
+    BYOS-2  SLES                   N/A     Do nothing
+    BYOS-2  SLES                   BYOS    Re-register
+    BYOS-2  BYOS-1                 N/A     De-register
```

Comment:
What are the actions taken by the user here? I do not understand this state transition in the context of this code. When a BYOS instance is registered, either to SCC or the update infrastructure then in order to transition from the registered state to the unregistered state `BYOS-2` to `BYOS-1` the user has to run `registercloudguest --clean`  meaning this code when it runs again will find a cache and no registration.

Reply:
True that the user has to run `registercloudguest --clean` command, but the code would check if license is BYOS,
which it would, it would check if it uses RMT as SCC proxy, which it wouldn't, then re-run the same command and
run the command to disable `guestregister`, no cache would be found on that scenario

If it did run again, after that scenario, yes there would be a cache saying `SLES_BYOS` on an unregistered system